src/xo-app/vm/tab-network.js

```diff
@@ -149,159 +145,226 @@ class VifItem extends BaseComponent {
+                  value={ip}
+                  xoType={resourceSet ? 'resourceSetIp' : 'ip'}
+                >
+                  {ip}
+                </XoSelect>
+              </Col>
+              <Col size={1}>
+                <ActionRowButton
+                  handler={this._deleteIp}
+                  handlerParam={ipIndex}
+                  icon='delete'
+                />
+              </Col>
+            </Row>
+          ))
+        )}
+        <Row>
+          <Col size={10}>
+            {showNewIpForm ? (
+              <span onBlur={this._toggleNewIp}>
+                {resourceSet ? (
+                  <SelectResourceSetIp
+                    autoFocus
+                    containerPredicate={this._getIsNetworkAllowed()}
+                    onChange={ip => this._addIp(ip)}
+                    predicate={this._getIpPredicate()}
+                    required
+                    resourceSetId={resourceSet}
+                  />
                 ) : (
-                  <ActionButton
-                    btnStyle='success'
-                    size='small'
-                    handler={this._toggleNewIp}
-                    icon='add'
+                  <SelectIp
+                    autoFocus
+                    containerPredicate={this._getIsNetworkAllowed()}
+                    onChange={ip => this._addIp(ip)}
+                    predicate={this._getIpPredicate()}
+                    required
                   />
                 )}
-              </Col>
-            </Row>
-          </Container>
-        </td>
-        <td>
+              </span>
+            ) : (
+              <ActionButton
+                btnStyle='success'
+                size='small'
+                handler={this._toggleNewIp}
+                icon='add'
+              />
+            )}
+          </Col>
+        </Row>
+      </Container>
+    )
+  }
+}
+
+const COLUMNS = [
+  {
+    itemRenderer: vif => `VIF #${vif.device}`,
+    name: _('vifDeviceLabel'),
+  },
+  {
+    itemRenderer: vif => (
+      <pre>
+        <Text value={vif.MAC} onChange={mac => setVif(vif, { mac })} />
+      </pre>
+    ),
+    name: _('vifMacLabel'),
+  },
+  {
+    itemRenderer: vif => vif.MTU,
+    name: _('vifMtuLabel'),
+  },
+  {
+    component: connectStore(() => {
+      const getVif = (state, props) => props.item
+      const getNetworkId = createSelector(getVif, vif => vif.$network)
+      const getNetwork = createGetObject(getNetworkId)
+
+      return (state, props) => ({
+        vif: getVif(state, props),
+        network: getNetwork(state, props),
+        networkId: getNetworkId(state, props),
+      })
+    })(({ vif, network, networkId }) => {
+      const _setNetwork = network => setVif(vif, { network })
+      const _getNetworkPredicate = createSelector(
+        () => vif && vif.$pool,
+        () => networkId,
+        (vifPoolId, vifNetworkId) => network =>
+          network.$pool === vifPoolId && network.id !== vifNetworkId
```

Comment:
I think we need to keep the current network in the list (ie remove `&& network.id !== vifNetworkId`) if we don't want the Select to show a red ID instead of the current network's name. Do you confirm this?